api/python/t4/api.py

```diff
@@ -101,6 +102,70 @@ def delete(target):
     delete_object(bucket, path)
 
 
+def _tophashes_with_packages(registry=None):
+    registry_base_path = get_package_registry(fix_url(registry) if registry else None)
+    registry_url = urlparse(registry_base_path)
+    if registry_url.scheme != 'file':
+        raise NotImplementedError
+
+    registry_dir = pathlib.Path(parse_file_url(registry_url))
+
+    out = {}
+    for pkg_namespace_path in (registry_dir / 'named_packages').iterdir():
+        pkg_namespace = pkg_namespace_path.name
+
+        for pkg_subname_path in pkg_namespace_path.iterdir():
+            pkg_subname = pkg_subname_path.name
+            pkg_name = pkg_namespace + '/' + pkg_subname
+
+            package_tophashes = [tophash.name for tophash in pkg_subname_path.iterdir()
+                                 if tophash.name != 'latest']
+
+            for tophash in package_tophashes:
+                if tophash in out:
+                    out[tophash].append(pkg_name)
+                else:
+                    out[tophash] = [pkg_name]
+
+    return out
+
+
+def delete_package(name, registry=None):
```

Comment:
There should be a summary of what all there is to be deleted with confirmation from the user, with a force=False argument.

Reply:
maybe call it skip_confirmation

Reply:
That's a good idea in spirit but nothing in the API right now takes user confirmation, and I don't really want to start doing that here right now.

Reply:
E.g. I think we should also consider having confirmation on `delete` and `delete_path` (which is a TODO), and that it should all be considered in a separate PR.

Reply:
I feel pretty strongly that we shouldn't ship this without a confirmation. @akarve please weigh in

Reply:
Well, we already don't ask for confirmation on `delete`.

Reply:
Can we leave that out of this PR and open a PR after this one for confirmation messages on `{delete, delete_dir, delete_package}`?

Reply:
I'm OK with that